ethportal-api/src/types/consensus/beacon_state.rs

```diff
@@ -0,0 +1,312 @@
+    #[superstruct(getter(copy))]
+    pub slot: u64,
+    #[superstruct(getter(copy))]
+    pub fork: Fork,
+
+    // History
+    pub latest_block_header: BeaconBlockHeader,
+    pub block_roots: FixedVector<H256, SlotsPerHistoricalRoot>,
+    pub state_roots: FixedVector<H256, SlotsPerHistoricalRoot>,
+    // Frozen in Capella, replaced by historical_summaries
+    pub historical_roots: VariableList<H256, HistoricalRootsLimit>,
+
+    // Ethereum 1.0 chain data
+    pub eth1_data: Eth1Data,
+    pub eth1_data_votes: VariableList<Eth1Data, SlotsPerEth1VotingPeriod>,
+    #[superstruct(getter(copy))]
+    #[serde(deserialize_with = "as_u64")]
+    pub eth1_deposit_index: u64,
+
+    // Registry
+    pub validators: VariableList<Validator, ValidatorRegistryLimit>,
+    pub balances: VariableList<u64, ValidatorRegistryLimit>,
+
+    // Randomness
+    pub randao_mixes: FixedVector<H256, EpochsPerHistoricalVector>,
+
+    // Slashings
+    pub slashings: FixedVector<u64, EpochsPerSlashingsVector>,
+
+    // Participation (Altair and later)
+    #[superstruct(only(Bellatrix, Capella))]
+    pub previous_epoch_participation: VariableList<ParticipationFlags, ValidatorRegistryLimit>,
+    #[superstruct(only(Bellatrix, Capella))]
+    pub current_epoch_participation: VariableList<ParticipationFlags, ValidatorRegistryLimit>,
+
+    // Finality
+    pub justification_bits: BitVector<JustificationBitsLength>,
+    #[superstruct(getter(copy))]
+    pub previous_justified_checkpoint: Checkpoint,
+    #[superstruct(getter(copy))]
+    pub current_justified_checkpoint: Checkpoint,
+    #[superstruct(getter(copy))]
+    pub finalized_checkpoint: Checkpoint,
+
+    // Inactivity
+    #[superstruct(only(Bellatrix, Capella))]
+    pub inactivity_scores: VariableList<u64, ValidatorRegistryLimit>,
+
+    // Light-client sync committees
+    #[superstruct(only(Bellatrix, Capella))]
+    pub current_sync_committee: Arc<SyncCommittee>,
+    #[superstruct(only(Bellatrix, Capella))]
+    pub next_sync_committee: Arc<SyncCommittee>,
+
+    // Execution
+    #[superstruct(
+        only(Bellatrix),
+        partial_getter(rename = "latest_execution_payload_header_bellatrix")
+    )]
+    pub latest_execution_payload_header: ExecutionPayloadHeaderBellatrix,
+    #[superstruct(
+        only(Capella),
+        partial_getter(rename = "latest_execution_payload_header_capella")
+    )]
+    pub latest_execution_payload_header: ExecutionPayloadHeaderCapella,
+
+    // Capella
+    #[superstruct(only(Capella), partial_getter(copy))]
+    #[serde(deserialize_with = "as_u64")]
+    pub next_withdrawal_index: u64,
+    #[superstruct(only(Capella), partial_getter(copy))]
+    #[serde(deserialize_with = "as_u64")]
+    pub next_withdrawal_validator_index: u64,
+    // Deep history valid from Capella onwards.
+    #[superstruct(only(Capella))]
+    pub historical_summaries: VariableList<HistoricalSummary, HistoricalRootsLimit>,
+}
+
+impl BeaconState {
+    #[cfg(test)]
```

Comment:
Really? Why do we only want to ssz decode the state inside our tests? Is this to avoid the `dead_code` lint? If so, then I'd prefer to use `allow[clippy::dead_code]`

Reply:
yes, it is to avoid the `dead_code` lint. I prefer this because that way we will not forget to remove the dead code attribute if it is not used.

Anyways, in #1167 I'm removing this.